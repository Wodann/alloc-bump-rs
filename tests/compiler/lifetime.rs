use alloc_bump::BumpAlloc;
use alloc_wg::alloc::Global;
use core::mem;

fn main() {
    let stack = 2.0f32;
    let heap = {
        let bump = BumpAlloc::<Global>::with_capacity_in(mem::size_of::<f32>(), Global);

        bump.alloc_t(stack).unwrap()
    };
    assert_eq!(stack, *heap);
}
