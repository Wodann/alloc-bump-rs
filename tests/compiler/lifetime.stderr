error[E0597]: `bump` does not live long enough
  --> $DIR/lifetime.rs:10:9
   |
10 |         bump.alloc_t(stack).unwrap()
   |         ^^^^------------------------
   |         |
   |         borrowed value does not live long enough
   |         borrow later used here
11 |     };
   |     - `bump` dropped here while still borrowed
